E0106 needs to be updated to new format

From: src/test/compile-fail/E0106.rs

Error E0106 needs a span_label, updating it from:

```
error[E0106]: missing lifetime specifier
  --> src/test/compile-fail/E0106.rs:12:8
   |
12 |     x: &bool, //~ ERROR E0106
   |        ^^^^^
```

To:

```
error[E0106]: missing lifetime specifier
  --> src/test/compile-fail/E0106.rs:12:8
   |
12 |     x: &bool, //~ ERROR E0106
   |        ^^^^^ expected lifetime information
```

@arielb1 Do you mean like this?
```
error[E0106]: missing lifetime specifier
  --> src/test/compile-fail/E0106.rs:12:8
   |
12 |     x: &bool, //~ ERROR E0106
   |        ^^^^^
   |        |
   |        expected lifetime parameter

```

Or like this, with the primary span changed?

```
error[E0106]: missing lifetime specifier
  --> src/test/compile-fail/E0106.rs:12:8
   |
12 |     x: &bool, //~ ERROR E0106
   |        ^ expected lifetime parameter

```

I think he means the second one:

```
error[E0106]: missing lifetime specifier
  --> src/test/compile-fail/E0106.rs:12:8
   |
12 |     x: &bool, //~ ERROR E0106
   |        ^ expected lifetime parameter

```

--- a/src/librustc_typeck/astconv.rs
+++ b/src/librustc_typeck/astconv.rs
@@ -310,8 +310,12 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
             None => match rscope.anon_regions(default_span, 1) {
                 Ok(rs) => rs[0],
                 Err(params) => {
-                    let mut err = struct_span_err!(self.tcx().sess, default_span, E0106,
-                                                   "missing lifetime specifier");
+                    let ampersand_span = Span { hi: default_span.lo, ..default_span};
+
+                    let mut err = struct_span_err!(self.tcx().sess, ampersand_span, E0106,
+                                                 "missing lifetime specifier");
+                    err.span_label(ampersand_span, &format!("expected lifetime parameter"));
+
                     if let Some(params) = params {
                         report_elision_failure(&mut err, params);
                     }

--- a/src/test/compile-fail/E0106.rs
+++ b/src/test/compile-fail/E0106.rs
@@ -9,13 +9,19 @@
 // except according to those terms.
 
 struct Foo {
-    x: &bool, //~ ERROR E0106
+    x: &bool,
+    //~^ ERROR E0106
+    //~| NOTE expected lifetime parameter
 }
 enum Bar {
     A(u8),
-    B(&bool), //~ ERROR E0106
+    B(&bool),
+   //~^ ERROR E0106
+   //~| NOTE expected lifetime parameter
 }
-type MyStr = &str; //~ ERROR E0106
+type MyStr = &str;
+        //~^ ERROR E0106
+        //~| NOTE expected lifetime parameter
 
 fn main() {
 }

